src/util/git.js

```diff
@@ -279,13 +287,21 @@ export default class Git {
     try {
       return await child.spawn('git', ['archive', `--remote=${this.url}`, this.ref, filename], {
         process(proc, update, reject, done) {
-          const parser = tar.Parse();
+          const parser = tarStream.extract();
 
           parser.on('error', reject);
-          parser.on('end', done);
-
-          parser.on('data', (entry: Buffer) => {
-            update(entry.toString());
+          parser.on('finish', done);
+
+          parser.on('entry', (header, stream, next) => {
+            let string = '';
+            stream.on('data', (buffer) => {
+              string += buffer.toString();
```

Comment:
iirc, this pattern might break when the buffer contains part of an unicode character but not the others. It might be better to concat the buffers together (possibly by storing them in a list and then concatenating everything at once with `Buffer.concat`) rather than concatenating strings.